src/test/java/com/palantir/ri/ResourceIdentifierTest.java

```diff
@@ -171,6 +171,18 @@ public void testStringConstruction() {
     }
 
     @Test
+    public void testStringConstructionWithMultipleLocatorComponents() {
+        assertEquals("ri.service..type.context.name1",
+                ResourceIdentifier.of("service", "", "type", "context", "name1", new String[0]).toString());
+        assertEquals("ri.service..type.context.name1.name2",
+                ResourceIdentifier.of("service", "", "type", "context", "name1",
+                        new String[] {"name2"}).toString());
+        assertEquals("ri.service..type.context.name1.name2.name3",
+                ResourceIdentifier.of("service", "", "type", "context", "name1",
+                        new String[] {"name2", "name3"}).toString());
```

Comment:
I suppose it would've been cool to test this with the varargs syntax instead of the array syntax